### PR TITLE
[Merged by Bors] - feat: define `MeasureTheory.OuterMeasureClass`

### PR DESCRIPTION
**Pull request by urkud**
---

This PR opens the road to generalizing theorems and definitions
from `μ : Measure α` to `[OuterMeasureClass F α] (μ : F)`.
None of these generalizations are included in this PR
to keep it small.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean [hunk 1]
```diff
@@ -94,6 +94,11 @@ instance Measure.instFunLike [MeasurableSpace α] : FunLike (Measure α) (Set α
 
 #noalign measure_theory.measure.has_coe_to_fun
 
+instance Measure.instOuterMeasureClass [MeasurableSpace α] : OuterMeasureClass (Measure α) α where
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: I'm wondering if you really want to declare this instance, or have a `MeasureClass` (that you would also instantiate for `FiniteMeasure`) and an instance that a `MeasureClass` is also an `OuterMeasureClass`?

Comment (urkud) [reply Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: `FiniteMeasure`s take values in `NNReal`s, not `ENNReal`s, see https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Measure/FiniteMeasure.html#MeasureTheory.FiniteMeasure.instCoeFun
AFAICT, `Measure` would be the only instance for `MeasureClass`, so I don't think that we should add this class.

Comment (sgouezel) [reply Mathlib/MeasureTheory/Measure/MeasureSpaceDef.lean:97]: Good point, I agree!

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**